manifests/install.pp

```diff
@@ -37,21 +36,73 @@
       'AIX'   => concat(['--ignoreos'],$install_options),
       default => $install_options
     }
-    $_package_options = {
+
+    # package provider does not provide 'versionable'
+    package { $::puppet_agent::package_name:
+      ensure          => 'present',
       provider        => 'rpm',
       source          => "/opt/puppetlabs/packages/${package_file_name}",
       install_options => $_install_options,
     }
   } elsif $::operatingsystem == 'Solaris' and $::operatingsystemmajrelease == '10' {
```

Comment:
So doing the upgrade does TERM the process that triggered the upgrade. Since Puppet then runs again and succeeds, and this is fixed in a future Puppet version, I'm tempted to ignore it (or at least treat it as a separate ticket).